driver/src/driver/scheduler.rs

```diff
@@ -0,0 +1,247 @@
+use super::stablex_driver::StableXDriver;
+use crate::metrics::StableXMetrics;
+use crate::orderbook::StableXOrderBookReading;
+use anyhow::{anyhow, Context, Result};
+use log::error;
+use log::info;
+use std::collections::HashSet;
+use std::thread;
+use std::time::{Duration, Instant, SystemTime, SystemTimeError};
+
+const BATCH_DURATION: Duration = Duration::from_secs(300);
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+struct BatchId(pub u64);
```

Comment:
Not a huge fan of all the `batch_id.0`. Should we just implement into<u64> and make this field private?

Reply:
Disagree. In this file we treat batch ids as u64 because it is more convenient than U256. Every u64 is a valid batch id and vice versa. The reason there is a wrapper is just to give this type a name and have one place to implement methods on it.

Reply:
True, doesn't have to be public and the pub did nothing anyway because the the wrapper itself wasn't public.